### ISSUE TITLE
Java error "out of bounds" in canonical decomposition

### ISSUE THREAD

**Issue by carlos-saez** [jdemetra/jdplus-main#301]
We have encountered a bug in JDemetra+ when performing the canonical decomposition of some ARIMA models. The problem appeared with two series in production in the v2 GUI, but still persists in v3. It seems to affect monthly series with ARIMA model (2,1,1)(0,1,0) and specific coefficients. 

The problem can be reproduced using the `rjd3toolkit` package with the following code:
```
model <- rjd3toolkit::sarima_model(period = 12, phi = c(0.183055, 0.0449044), d = 1, bd = 1, theta = -0.499883)
rjd3toolkit::sarima_decompose(model)
```
The error we get is:
```
Error en .jcall("jdplus/toolkit/base/r/arima/UcarimaModels", "[B", "toBuffer", : 
  java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
```

**Comment by Immurb** [comment]
This is the stacktrace I get:

`java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at internal.toolkit.base.core.math.linearfilters.SymmetricMullerNewtonSolver.update(SymmetricMullerNewtonSolver.java:876)
	at internal.toolkit.base.core.math.linearfilters.SymmetricMullerNewtonSolver.newtonnull(SymmetricMullerNewtonSolver.java:546)
	at internal.toolkit.base.core.math.linearfilters.SymmetricMullerNewtonSolver.factorize(SymmetricMullerNewtonSolver.java:336)
	at internal.toolkit.base.core.math.linearfilters.RobustSymmetricFrequencyResponseDecomposer.decompose(RobustSymmetricFrequencyResponseDecomposer.java:104)
	at internal.toolkit.base.core.math.linearfilters.SymmetricFilterAlgorithms.lambda$robustFactorizer$2(SymmetricFilterAlgorithms.java:92)
	at internal.toolkit.base.core.math.linearfilters.SymmetricFilterAlgorithms.lambda$factorizer$6(SymmetricFilterAlgorithms.java:143)
	at jdplus.toolkit.base.core.math.linearfilters.SymmetricFilter.factorize(SymmetricFilter.java:469)
	at jdplus.toolkit.base.core.arima.ArimaModel.getMa(ArimaModel.java:341)
	at jdplus.toolkit.base.protobuf.modelling.ModellingProtosUtility.convert(ModellingProtosUtility.java:243)
	at jdplus.toolkit.base.r.arima.UcarimaModels.toBuffer(UcarimaModels.java:154)`

**Comment by palatej** [comment]
That's a numerical issue in some limit cases. Corrected in #321
